src/Engine/RadiumEngine.hpp

```diff
@@ -198,6 +211,20 @@ class RA_ENGINE_API RadiumEngine
 
     /// For internal resources management in a filesystem
     std::string m_resourcesRootDir;
+
+    /// active fbo and viewport management, this class handle fbo and viewport
+    /// values so that one can save and restore these value with pushFboAndViewport and popFboAndViewport
+    class FboAndViewport
+    {
+      public:
+        FboAndViewport( int fbo, int viewport[4] ) :
+            m_fbo{fbo}, m_viewport{viewport[0], viewport[1], viewport[2], viewport[3]} {}
+        int m_fbo;
+        int m_viewport[4];
+    };
+
+    /// Stack of saved fbo and viewport values @see pushFboAndViewport popFboAndViewport
+    std::vector<FboAndViewport> m_fboAndViewportStack;
```

Comment:
if this is a stack, why not using `std::stack` instead of `std::vector`. By default, `std::stack` uses `std::deque`container, that gives a push/pop complexity in O(1) where `std::vector` in only amortized O(1).